srcpkgs/micropython/template

```diff
@@ -9,10 +9,11 @@ short_desc="Implementation of Python 3.x for microcontrollers"
 maintainer="Orphaned <orphan@voidlinux.org>"
 license="MIT"
 homepage="https://micropython.org/"
-distfiles="https://github.com/micropython/micropython/releases/download/v${version}/micropython-${version}.tar.gz"
-checksum=9ab32eb2c19a682f6de7e9b4226de849c42cb92e1ecb4cf56438a75d763bd451
+distfiles="https://github.com/micropython/micropython/releases/download/v${version}/micropython-${version}.tar.xz"
+checksum=97306156fdeab120a1244626c75a929bb820722afdfc1317dbd5dadef388d94c
 
 case "$XBPS_TARGET_MACHINE" in
+	armv5tel-musl) nopie=yes ;;
```

Comment:
This might (almost definitely) be a bug you want to report to musl folks. Can you show a stack trace / disassembly output?

Reply:
@lfdebrux can you check if their build system doesn't simply disable PIC? By passing `-fno-PIC` to the compiler, for example. If that's the case, what can happen is that it interacts badly with our PIE by default toolchain, and creates a text relocation in the binary. Musl doesn't support those at all.

You can check for their presence with `readelf -d <binary>` and seeing if TEXTREL appears in the output.

Reply:
It doesn't look like they disable PIC; what's weird is that it works fine on x86_64-musl with PIE, but doesn't on armv5tel-musl, which as you say does point towards it being a bug in musl.

Here's a stacktrace from gdb

```
(gdb) start
Temporary breakpoint 1 at 0x90f0: file main.c, line 421.
Starting program: /usr/bin/micropython

Program received signal SIGSEGV, Segmentation fault.
0xb6fb68c0 in do_relocs () from /lib/ld-musl-arm.so.1
(gdb) backtrace
#0  0xb6fb68c0 in do_relocs () from /lib/ld-musl-arm.so.1
#1  0xb6fb6c88 in reloc_all () from /lib/ld-musl-arm.so.1
#2  0xb6fb900c in __dls3 () from /lib/ld-musl-arm.so.1
#3  0xb6fb867c in __dls2 () from /lib/ld-musl-arm.so.1
#4  0xb6fb569c in _dlstart () from /lib/ld-musl-arm.so.1
Backtrace stopped: previous frame identical to this frame (corrupt stack?)
(gdb)
```

Reply:
Seems it also segfaults on armv6l :P 

I will try to determine which platforms we have to enable `nopie` for.

Reply:
> 0x00000016 (TEXTREL)                    0x0

As suspected :) So not really a bug on musl's side, though it would be nice if an error message were added.

This might be due to their assembly code somewhere, though I couldn't find it. Weirdly, Alpine disables `armv7l` as a target because tests don't pass, and even `armhf` (which is equivalent to our armv6l) because it doesn't build (it might simply be outdated, of course).

Reply:
Oh interesting. With `nopie` that isn't present. Sorry I can't be more helpful, I'm a bit out my depth!